main.py

```diff
@@ -648,23 +648,105 @@ def on_link_down(self, event):
 
     def handle_link_down(self, event):
         """Change circuit when link is down or under_mantenance."""
-        log.debug("Event handle_link_down %s", event)
+        link = event.content["link"]
+        log.info("Event handle_link_down %s", link)
+        switch_flows = {}
+        evcs_with_failover = []
+        evcs_normal = []
+        check_failover = []
         for evc in self.circuits.values():
+            if evc.is_affected_by_link(link):
```

Comment:
Thanks for your observation and suggestion, @viniarck ! I'm glad you caughted this before we merge the PR. I've pushed commit cf543fd to fix this following your suggestion (using shallow copy for now). In the future, we can refactor the solution (to use Mongo queries). Since you already mapped Issue #177, let's use this shallow copy as a workaround for now.

Reply:
Right. Looks good to me @italovalcy. Feel free to close that issue when this PR gets merged, thanks